### PR TITLE
Spanish translation of the file training.mdx

### PR DESCRIPTION
**Pull request by yharyarias**
I made the translation of the **training.mdx** file in collaboration to the transformers/doc/ documentation, there I created a new folder called source_es where I hosted my training.mdx document translated to Spanish, contributing to the Spanish-speaking community. 
[forum](https://github.com/huggingface/transformers/issues/15947)

Approved by @omarespejel
Documentation: @sgugger

### CODE REVIEW

File: docs/source_es/Hugging.mdx [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+Note: Cada dataset debería ir acompañado de un [Dataset Card](https://huggingface.co/docs/datasets/v1.12.0/dataset_card.html) para promover el uso responsable, y alertar al usuario de cualquier posible sesgo dentro del conjunto de datos.
+
+```python
+from transformers import AutoModelForSequenceClassification
+
+model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+> Verás una advertencia acerca de que algunos de los pesos pre-entrenados que no están siendo utilizados y que algunos pesos están siendo inicializados al azar. No te preocupes, esto es completamente normal. El head/cabezal pre-entrenado del modelo BERT se descarta y se sustituye por un head de clasificación inicializado aleatoriamente. Puedes aplicar fine-tuning a este nuevo head del modelo en su tarea de clasificación de secuencias, haciendo transfer learning del modelo pre-entrenado.
+>
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:78]: This empty > is causing some issues.
```suggestion
```

File: docs/source_es/Hugging.mdx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Haciendo fine-tuning a un modelo pre-entrenado
+
+El uso de un modelo pre-entrenado tiene importantes ventajas. Reduce los costes de computación, su huella de carbono, y te permite utilizar modelos de última generación sin tener que entrenar uno desde cero. 🤗 Transformers proporciona acceso a miles de modelos pre-entrenados en una amplia gama de tareas. Cuando utilizas un modelo pre-entrenado, lo entrenas con un dataset específico para tu tarea. Esto se conoce como fine-tuning, una técnica de entrenamiento increíblemente poderosa. En este tutorial haremos fine-tuning a un modelo pre-entrenado con un framework de Deep Learning de tu elección:
+
+- Aplicar fine-tuning a un modelo pre-entrenado con 🤗 Transformers [Trainer](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.Trainer).
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:17]:
```suggestion
- Aplicar fine-tuning a un modelo pre-entrenado con 🤗 Transformers [`Trainer`].
```
We shouldn't hardcode any link like this, as the version will never change.

File: docs/source_es/Hugging.mdx [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+```python
+small_train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(1000))
+small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000))
+```
+
+## Fine-tuning con [Trainer]
+
+[https://youtu.be/nvBXf7s7vTI](https://youtu.be/nvBXf7s7vTI)
+
+🤗 Transformers proporciona una clase **[Trainer**](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.Trainer) optimizada para el entrenamiento de modelos 🤗 Transformers, haciendo más fácil el inicio del entrenamiento sin necesidad de escribir manualmente su propio ciclo. La API del [Trainer](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.Trainer) soporta una amplia gama de opciones de entrenamiento y características como el logging, el gradient accumulation y la mixed precision.
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:65]:
```suggestion
🤗 Transformers proporciona una clase [`Trainer`] optimizada para el entrenamiento de modelos 🤗 Transformers, haciendo más fácil el inicio del entrenamiento sin necesidad de escribir manualmente su propio ciclo. La API del [`Trainer`] soporta una amplia gama de opciones de entrenamiento y características como el logging, el gradient accumulation y la mixed precision.
```

File: docs/source_es/Hugging.mdx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+```
+
+Si lo deseas, puedes crear un subconjunto más pequeño del dataset completo para aplicarle fine-tuning y así reducir el tiempo.
+
+```python
+small_train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(1000))
+small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000))
+```
+
+## Fine-tuning con [Trainer]
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:61]:
```suggestion
## Fine-tuning con `Trainer`
```

File: docs/source_es/Hugging.mdx [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+
+[https://youtu.be/nvBXf7s7vTI](https://youtu.be/nvBXf7s7vTI)
+
+🤗 Transformers proporciona una clase **[Trainer**](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.Trainer) optimizada para el entrenamiento de modelos 🤗 Transformers, haciendo más fácil el inicio del entrenamiento sin necesidad de escribir manualmente su propio ciclo. La API del [Trainer](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.Trainer) soporta una amplia gama de opciones de entrenamiento y características como el logging, el gradient accumulation y la mixed precision.
+
+Comienza cargando tu modelo y especifica el número de labels previstas. A partir del Card Dataset Yelp Review, que como ya sabemos tiene 5 labels:
+
+Note: Cada dataset debería ir acompañado de un [Dataset Card](https://huggingface.co/docs/datasets/v1.12.0/dataset_card.html) para promover el uso responsable, y alertar al usuario de cualquier posible sesgo dentro del conjunto de datos.
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:69]: This is different from the English version.

File: docs/source_es/Hugging.mdx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+```python
+from transformers import AutoModelForSequenceClassification
+
+model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+> Verás una advertencia acerca de que algunos de los pesos pre-entrenados que no están siendo utilizados y que algunos pesos están siendo inicializados al azar. No te preocupes, esto es completamente normal. El head/cabezal pre-entrenado del modelo BERT se descarta y se sustituye por un head de clasificación inicializado aleatoriamente. Puedes aplicar fine-tuning a este nuevo head del modelo en su tarea de clasificación de secuencias, haciendo transfer learning del modelo pre-entrenado.
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:77]: Use the <Tip> syntax as in the English version.

File: docs/source_es/Hugging.mdx [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+>
+
+## Hiperparámetros de entrenamiento
+
+A continuación, crea una clase [TrainingArguments](https://huggingface.co/docs/transformers/v4.17.0/en/main_classes/trainer#transformers.TrainingArguments) que contenga todos los hiperparámetros que puedes ajustar, así como los indicadores para activar las diferentes opciones de entrenamiento. Para este tutorial puedes empezar con los [hiperparámetros](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) de entrenamiento por defecto, pero siéntete libre de experimentar con ellos para encontrar tu configuración óptima.
```
Comment (sgugger) [inline docs/source_es/Hugging.mdx:82]:
```suggestion
A continuación, crea una clase [`TrainingArguments`] que contenga todos los hiperparámetros que puedes ajustar, así como los indicadores para activar las diferentes opciones de entrenamiento. Para este tutorial puedes empezar con los [hiperparámetros](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) de entrenamiento por defecto, pero siéntete libre de experimentar con ellos para encontrar tu configuración óptima.
```

**Review by sgugger — CHANGES_REQUESTED** [review]
I've just reviewed the beginning but there are a lot of differences with the English version of this doc in terms of syntax used. Could you keep the same syntax for tips, links to other classes in the doc etc?

**Comment by yharyarias** [comment]
@sgugger Of course, I will make the changes you indicate, leaving the syntax as per the original documentation. I have a question, I see that most of the conflicts are because of the links, do I leave all the links or remove them? I think they are important.

**Comment by sgugger** [comment]
The links are automatically generated by our doc building tool. You should leave them as they are in the `training.mdx` file English version (for instance [`Trainer`]).

**Comment by yharyarias** [comment]
I just uploaded the changes you suggested @sgugger I'll be watching for your review.

**Review by sgugger — APPROVED** [review]
Thanks a lot that's way better! You can remove the `Hugging.mdx` file from your PR.
I can't judge the translation itself but the format of the file is very good :-)

**Comment by omarespejel** [comment]
Thanks! Will review the translation.

**Review by LysandreJik — APPROVED** [review]
This is great, thanks for working on it @yharyarias!

File: docs/source_es/training.mdx [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+
+Crea un optimizador y el learning rate para aplicar fine-tuning al modelo. Vamos a utilizar el optimizador [`AdamW`](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html) de PyTorch:
+
+```py
+>>> from torch.optim import AdamW
+
+>>> optimizer = AdamW(model.parameters(), lr=5e-5)
+```
+
+Crea el learning rate por defecto desde el [`Trainer`]:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:287]:
```suggestion
Crea el learning rate desde el [`Trainer`]:
```

File: docs/source_es/training.mdx [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+A continuación, crea un subconjunto más pequeño del dataset, como se ha mostrado anteriormente, para acelerar el fine-tuning:
+
+```py
+>>> small_train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(1000))
+>>> small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000))
+```
+
+### DataLoader
+
+Crea un `DataLoader` para sus datasets de entrenamiento y de prueba para poder iterar sobre batches de datos:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:260]:
```suggestion
Crea un `DataLoader` para tus datasets de entrenamiento y de prueba para poder iterar sobre batches de datos:
```
Aquí por favor siempre usar la segunda persona del singular ya que lo hemos estado usando durante todo el doc

File: docs/source_es/training.mdx [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+En este punto, es posible que necesites reiniciar tu notebook o ejecutar el siguiente código para liberar algo de memoria:
+
+```py
+del model
+del pytorch_model
+del trainer
+torch.cuda.empty_cache()
+```
+
+A continuación, haz un post-proceso manualmente al `tokenized_dataset` y así prepararlo para el entrenamiento.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:231]:
```suggestion
A continuación, haremos un post-procesamiento manual al `tokenized_dataset` y así prepararlo para el entrenamiento.
```

File: docs/source_es/training.mdx [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+>>> model.fit(tf_train_dataset, validation_data=tf_validation_dataset, epochs=3)
+```
+
+<a id='pytorch_native'></a>
+
+## Fine-tune en PyTorch nativo
+
+<Youtube id="Dh9CL8fyG80"/>
+
+El [`Trainer`] se encarga del ciclo de entrenamiento y permite aplicar fine-tuning a un modelo en una sola línea de código. Para los usuarios que prefieren escribir su propio ciclo de entrenamiento, también puedes aplicar fine-tuning a un modelo 🤗 Transformers en PyTorch nativo.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:220]:
```suggestion
El [`Trainer`] se encarga del ciclo de entrenamiento y permite aplicar fine-tuning a un modelo en una sola línea de código. Para los usuarios que prefieren escribir su propio ciclo de entrenamiento, también puedes aplicar fine-tuning a un modelo de 🤗 Transformers en PyTorch nativo.
```

File: docs/source_es/training.mdx [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+Carguemos un modelo TensorFlow con el número esperado de labels:
+
+```py
+>>> import tensorflow as tf
+>>> from transformers import TFAutoModelForSequenceClassification
+
+>>> model = TFAutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+A continuación, compila y aplica fine-tuning a tu modelo con [`fit`](https://keras.io/api/models/model_training_apis/) como lo harías con cualquier otro modelo Keras:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:202]:
```suggestion
A continuación, compila y aplica fine-tuning a tu modelo con [`fit`](https://keras.io/api/models/model_training_apis/) como lo harías con cualquier otro modelo de Keras:
```

File: docs/source_es/training.mdx [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+>>> data_collator = DefaultDataCollator(return_tensors="tf")
+```
+
+<Tip>
+
+[`Trainer`] utiliza [`DataCollatorWithPadding`] por defecto, por lo que no es necesario especificar explícitamente un intercalador de datos.
+
+</Tip>
+
+A continuación, convierte los datasets0 tokenizados en datasets de TensorFlow con el método [`to_tf_dataset`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.to_tf_dataset). Especifica tus entradas en `columns`, y tu etiqueta en `label_cols`:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:171]:
```suggestion
A continuación, convierte los datasets tokenizados en datasets de TensorFlow con el método [`to_tf_dataset`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.to_tf_dataset). Especifica tus entradas en `columns` y tu etiqueta en `label_cols`:
```

File: docs/source_es/training.mdx [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+
+```py
+>>> from transformers import DefaultDataCollator
+
+>>> data_collator = DefaultDataCollator(return_tensors="tf")
+```
+
+<Tip>
+
+[`Trainer`] utiliza [`DataCollatorWithPadding`] por defecto, por lo que no es necesario especificar explícitamente un intercalador de datos.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:167]:
```suggestion
[`Trainer`] utiliza [`DataCollatorWithPadding`] por defecto por lo que no es necesario especificar explícitamente un intercalador de datos (data collator, en inglés).
```

File: docs/source_es/training.mdx [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+>>> trainer.train()
+```
+
+<a id='keras'></a>
+
+## Fine-tuning con Keras
+
+<Youtube id="rnTGBy2ax1c"/>
+
+Los modelos de 🤗 Transformers también permite realizar el entrenamiento en TensorFlow con la API de Keras. Sólo es necesario hacer algunos cambios antes de hacer fine-tuning.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:153]:
```suggestion
Los modelos de 🤗 Transformers también permiten realizar el entrenamiento en TensorFlow con la API de Keras. Sólo es necesario hacer algunos cambios antes de hacer fine-tuning.
```

File: docs/source_es/training.mdx [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     train_dataset=small_train_dataset,
+...     eval_dataset=small_eval_dataset,
+...     compute_metrics=compute_metrics,
+... )
+```
+
+A continuación, aplique fine-tuning a su modelo llamando a [`~transformers.Trainer.train`]:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:141]:
```suggestion
A continuación, aplique fine-tuning a su modelo llamando [`~transformers.Trainer.train`]:
```

File: docs/source_es/training.mdx [hunk 17]
```diff
@@ -0,0 +1,367 @@
...
+
+```py
+>>> from transformers import TrainingArguments
+
+>>> training_args = TrainingArguments(output_dir="test_trainer", evaluation_strategy="epoch")
+```
+
+### Trainer
+
+Crea un objeto [`Trainer`] con su modelo, argumentos de entrenamiento, conjuntos de datos de entrenamiento y de prueba, y su función de evaluación:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:129]:
```suggestion
Crea un objeto [`Trainer`] con tu modelo, argumentos de entrenamiento, conjuntos de datos de entrenamiento y de prueba, y tu función de evaluación:
```

File: docs/source_es/training.mdx [hunk 18]
```diff
@@ -0,0 +1,367 @@
...
+```
+
+Como ya sabes, necesitas un tokenizador para procesar el texto e incluir una estrategia para el padding y el truncamiento, manejando cualquier tamaño de una variable secuencial. Para procesar tu dataset en un solo paso, utiliza el método 🤗 Datasets [`map`](https://huggingface.co/docs/datasets/process.html#map) para aplicar una función de preprocesamiento sobre todo el dataset:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:42]:
```suggestion
Como ya sabes, necesitas un tokenizador para procesar el texto e incluir una estrategia para el padding y el truncamiento para manejar cualquier longitud de secuencia variable. Para procesar tu dataset en un solo paso utiliza el método de 🤗 Datasets [`map`](https://huggingface.co/docs/datasets/process.html#map) para aplicar una función de preprocesamiento sobre todo el dataset:
```

File: docs/source_es/training.mdx [hunk 19]
```diff
@@ -0,0 +1,367 @@
...
+>>> small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000))
+```
+
+<a id='trainer'></a>
+
+## Fine-tuning con `Trainer`
+
+<Youtube id="nvBXf7s7vTI"/>
+
+🤗 Transformers proporciona una clase [`Trainer`] optimizada para el entrenamiento de modelos 🤗 Transformers, haciendo más fácil el inicio del entrenamiento sin necesidad de escribir manualmente su propio ciclo. La API del [`Trainer`] soporta una amplia gama de opciones de entrenamiento y características como el logging, el gradient accumulation y el mixed precision.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:70]:
```suggestion
🤗 Transformers proporciona una clase [`Trainer`] optimizada para el entrenamiento de modelos 🤗 Transformers, haciendo más fácil el inicio del entrenamiento sin necesidad de escribir manualmente tu propio ciclo. La API del [`Trainer`] soporta una amplia gama de opciones de entrenamiento y características como el logging, el gradient accumulation y el mixed precision.
```

File: docs/source_es/training.mdx [hunk 20]
```diff
@@ -0,0 +1,367 @@
...
+
+Comienza cargando tu modelo y especifica el número de labels previstas. A partir del [Card Dataset](https://huggingface.co/datasets/yelp_review_full#data-fields) Yelp Review, que como ya sabemos tiene 5 labels:
```
Comment (omarespejel) [inline docs/source_es/training.mdx:72]:
```suggestion
Comienza cargando tu modelo y especifica el número de labels previstas. A partir del [Card Dataset](https://huggingface.co/datasets/yelp_review_full#data-fields) de Yelp Review, que como ya sabemos tiene 5 labels:
```

File: docs/source_es/training.mdx [hunk 21]
```diff
@@ -0,0 +1,367 @@
...
+<Tip>
+
+Verás una advertencia acerca de que algunos de los pesos pre-entrenados que no están siendo utilizados y que algunos pesos están siendo inicializados al azar. 
+No te preocupes, esto es completamente normal. El head/cabezal pre-entrenado del modelo BERT se descarta y se sustituye por un head de clasificación inicializado aleatoriamente. Puedes aplicar fine-tuning a este nuevo head del modelo en su tarea de clasificación de secuencias, haciendo transfer learning del modelo pre-entrenado.
+
+</Tip>
+
+### Hiperparámetros de entrenamiento
+
+A continuación, crea una clase [`TrainingArguments`] que contenga todos los hiperparámetros que puedes ajustar, así como los indicadores para activar las diferentes opciones de entrenamiento. Para este tutorial puedes empezar con los [hiperparámetros](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) de entrenamiento por defecto, pero siéntete libre de experimentar con ellos para encontrar tu configuración óptima.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:89]:
```suggestion
A continuación, crea una clase [`TrainingArguments`] que contenga todos los hiperparámetros que puedes ajustar así como los indicadores para activar las diferentes opciones de entrenamiento. Para este tutorial puedes empezar con los [hiperparámetros](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) de entrenamiento por defecto, pero siéntete libre de experimentar con ellos para encontrar tu configuración óptima.
```

File: docs/source_es/training.mdx [hunk 22]
```diff
@@ -0,0 +1,367 @@
...
+```py
+>>> from transformers import AutoModelForSequenceClassification
+
+>>> model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+<Tip>
+
+Verás una advertencia acerca de que algunos de los pesos pre-entrenados que no están siendo utilizados y que algunos pesos están siendo inicializados al azar. 
+No te preocupes, esto es completamente normal. El head/cabezal pre-entrenado del modelo BERT se descarta y se sustituye por un head de clasificación inicializado aleatoriamente. Puedes aplicar fine-tuning a este nuevo head del modelo en su tarea de clasificación de secuencias, haciendo transfer learning del modelo pre-entrenado.
```
Comment (omarespejel) [inline docs/source_es/training.mdx:83]:
```suggestion
No te preocupes, esto es completamente normal. El head/cabezal pre-entrenado del modelo BERT se descarta y se sustituye por un head de clasificación inicializado aleatoriamente. Puedes aplicar fine-tuning a este nuevo head del modelo en tu tarea de clasificación de secuencias haciendo transfer learning del modelo pre-entrenado.
```

**Review by omarespejel — CHANGES_REQUESTED** [review]
Muy bueno @yharyarias! Thank you! I added some comments

**Comment by yharyarias** [comment]
I made the last changes suggested by @omarespejel , I think everything is ok, this is my first contribution and there will be more to come, I have loved collaborating with open source. Thank you very much 🤗 @sgugger @LysandreJik  and @omarespejel

**Comment by omarespejel** [comment]
Thanks, @yharyarias! There are still a couple of comments unresolved.

**Review by omarespejel — APPROVED** [review]

**Comment by omarespejel** [comment]
IMO it is good to go :+1:

**Comment by sgugger** [comment]
Thanks again for your PR. Merging this but we will wait a little bit more to serve a doc in Spanish mainly:
- to have a few more pages
- to automatically redirect to English doc when the page is not available in Spanish.

I will put a comment here when the page is live (hopefully in a week or two!)

**Comment by yharyarias** [comment]
@sgugger Thank you, of course I will continue to contribute, I'm currently working on the translation of another document to add to source_es.

**Pull request merged**